src/OrbitGl/TrackHeader.cpp

```diff
@@ -119,25 +91,34 @@ void orbit_gl::TrackHeader::DoUpdateLayout() {
 }
 
 void TrackHeader::UpdateCollapseToggle() {
-  const float label_height = layout_->GetTrackTabHeight();
-  const float half_label_height = 0.5f * label_height;
-  const float x0 = GetPos()[0] + layout_->GetTrackTabOffset() +
-                   layout_->GetTrackIndentOffset() * track_->GetIndentationLevel();
-  const float button_offset = layout_->GetCollapseButtonOffset();
+  const float x0 = GetPos()[0] + layout_->GetTrackIndentOffset() * track_->GetIndentationLevel();
   const float size = layout_->GetCollapseButtonSize(track_->GetIndentationLevel());
-  const float toggle_y_pos = GetPos()[1] + half_label_height - size / 2;
-  Vec2 toggle_pos = Vec2(x0 + button_offset, toggle_y_pos);
+
+  constexpr float kOffsetX = -0.5f;
+  constexpr float kOffsetY = 1.f;
+  const float toggle_x_pos = x0 + layout_->GetCollapseButtonOffset() + kOffsetX;
+  const float toggle_y_pos =
+      GetPos()[1] + layout_->GetTextBoxHeight() * 0.5f - size * 0.5f + kOffsetY;
+
+  Vec2 toggle_pos = Vec2(toggle_x_pos, toggle_y_pos);
 
   collapse_toggle_->SetWidth(size);
   collapse_toggle_->SetHeight(size);
-  collapse_toggle_->SetPos(toggle_pos[0], toggle_pos[1]);
+  collapse_toggle_->SetPos(toggle_pos[0], toggle_pos[1] + GetVerticalLabelOffset());
 
   // Update the "collapsible" property of the triangle toggle to match the same property in the
   // parent track. This makes sure that changes to the track "collapsible" property are correctly
   // disabling the triangle toggle, even if they change during runtime.
   collapse_toggle_->SetIsCollapsible(track_->IsCollapsible());
 }
 
+float TrackHeader::GetVerticalLabelOffset() const {
```

Comment:
This only works when IndentationLevel is 1. Please solve the issue or include that information in the comment and log an error in case that IndentationLevel is higher than 1.

Reply:
Added a log as fixing this properly is a change that will come in its own PR, basically overhauling the track hierarchy system.